apps/microsoft-teams/app-actions/src/actions/list-channels.ts

```diff
@@ -1,30 +1,39 @@
 import { AppActionCallContext } from '@contentful/node-apps-toolkit';
-import { AppActionCallResponse, Channel } from '../types';
+import { Channel, AppActionCallResponse } from '../types';
 import { fetchTenantId } from '../utils';
 import helpers from '../helpers';
-import { withAsyncAppActionErrorHandling } from '../helpers/error-handling';
+import { ApiError } from '../errors';
 
-export const handler = withAsyncAppActionErrorHandling(
```

Comment:
I'm _somewhat_ partial to this handler wrapper (probably because I wrote it 😬 ) but that's mainly because it "guarantees" a result type response to be returned from the handler (ie since all unhandled exceptions are caught and recast to that object.

With the refactor we lose that guarentee -- eg if fetchTenantId fails for unknown reasons then an unexpected result is returned.

This isn't necessarily a bad thing -- because now we have this amazing new result type behavior baked into our API. However, that implies two things:

1. We would need to adapt the upstream consumer(s) of this app action to work with that new result type.
2. The addition of the try/catch and the unknown error we raise there would not be needed. We'd just want to let the unhandled exception bubble up.